runtime/jcl/common/sun_misc_Unsafe.cpp

```diff
@@ -788,6 +788,58 @@ Java_jdk_internal_misc_Unsafe_objectFieldOffset1(JNIEnv *env, jobject receiver,
 	return offset;
 }
 
+void JNICALL
+Java_jdk_internal_misc_Unsafe_writebackMemory(JNIEnv *env, jobject receiver, jlong addr, jlong len)
+{
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+
+#if defined(LINUX) && (defined(J9X86) || defined(J9HAMMER))
```

Comment:
How expensive is the work to find the cacheline / check the feature description?  If the goal is to write the memory back efficiently, we need to be careful not to overwhelm the benefits by doing too much work on each call

Reply:
It appears that calling CPUID is a serializing instruction. I haven't been able to find latency numbers on any modern processors but stackoverflow seems to suggest it is a fairly heavy operation. I could store feature info in the Unsafe java class and then the native `writebackMemory` function would only need to call the cache writeback instructions for the address range.